conceptofmind/LaMDA-pytorch
# Add poetry packaging

# Description

This PR is focused on the TODO note: `Add pip installer with PyPI`. With the enablement of packaging with [poetry](https://python-poetry.org/). The dependencies are added in `pyproject.toml` along with the project description, the author's email has to be added afterward.

# Testing Steps
1. Clone the repo
2. Install poetry. See official [guide](https://python-poetry.org/docs/#installation)
3. Navigate to the directory
4. Install the dependencies `poetry install`
5. Activate the virtual environment `poetry shell`
6. Try:
```
from lamda_pytorch.lamda_pytorch import LaMDA
from lamda_pytorch.utils.utils import AutoregressiveWrapper
import torch

if __name__ == "__main__":
    lamda_base = LaMDA(
        num_tokens = 20000,
        dim = 512,
        dim_head = 64,
        depth = 12,
        heads = 8
    )

    lamda = AutoregressiveWrapper(lamda_base, max_seq_len = 512)

    tokens = torch.randint(0, 20000, (1, 512)) # mock token data

    logits = lamda(tokens)

    print(logits)
```

# Publishing

Poetry supports publishing to PyPi the packaged library, for that follow the official [guide](https://python-poetry.org/docs/libraries/#:~:text=Poetry%20will%20publish%20to%20PyPI,publish%20on%20PyPI%20as%20well.).

1. Build your library `poetry build`
2. Publish it `poetry publish`
